engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java

```diff
@@ -1,63 +1,163 @@
+/*
+ * Copyright 2018-Present Entando S.r.l. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
 package org.entando.entando.aps.system.services.oauth2;
 
 import com.agiletec.aps.system.common.AbstractService;
 import com.agiletec.aps.system.exception.ApsSystemException;
+import java.util.Calendar;
+import java.util.Collection;
 import org.entando.entando.aps.system.services.oauth2.model.OAuth2Token;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.entando.entando.aps.system.services.oauth2.model.OAuth2AccessTokenImpl;
+import org.springframework.security.oauth2.common.DefaultOAuth2RefreshToken;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2RefreshToken;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
 
 public class ApiOAuth2TokenManager extends AbstractService implements IApiOAuth2TokenManager {
 
     private static final Logger logger = LoggerFactory.getLogger(ApiOAuth2TokenManager.class);
     private static final String ERROR_ADDING_TOKEN = "Error adding OAuth2Token";
     private transient final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
 
-    private OAuth2TokenDAO oAuth2TokenDAO;
+    private IOAuth2TokenDAO oAuth2TokenDAO;
 
-    public OAuth2TokenDAO getOAuth2TokenDAO() {
+    @Override
```

Comment:
Test coverage for the new methods.

Reply:
Added tests for TokenDao and TokenManager